<launch>
    <node name="control" pkg="hw3" type="control.py" output="screen">
      <param name="linear_speed" value="$(arg linear_speed)"/>
      <param name="angular_speed" value="$(arg angular_speed)"/>
    </node>
    <node name="sensor" pkg="hw3" type="sensor.py" output="screen"/>

  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>
  <arg name="yaw" default="0.75"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/detect_obstacles.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos)  -Y $(arg yaw) -param robot_description" />

  <node pkg="hw3" type="monitor.py" name="monitor"></node>
  <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_gazebo_rviz.launch"/>

</launch>

<!-- roslaunch hw3 obstacle_ditect.launch angular_speed:=0.2 linear_speed:=0.2 -->